File: Python/BasicMaths/Maths.py
print("Running Maths in Python")
n = int(input("Enter the value of n"))

def count_digits(n) :
    count = 0
    while n > 0 : 
        ld = n %10
        count += 1
        n //= 10
    return count;

def reverse_a_number(n) : 
    num = 0;
    while n > 0 :
        lastDigit = n % 10
        num = num*10 + lastDigit
        n //=10
    return num;

def check_palindrome(n) : 
    return True if n == reverse_a_number(n) else False ;

def armstrong_number(n) : 
    sum = 0, num = n
    while n > 0 :
        lastDigit = n % 10
        sum += pow(lastDigit,3)
        n //=10
    return True if n == sum else False;

import math

def print_all_divisors(n):
    if n <= 0:
        return []
        
    arr = []
    for i in range(1, int(math.sqrt(n)) + 1):
        if n % i == 0: 
            arr.append(i)
            if i*i != n: 
                arr.append(n // i)
    
    arr.sort()
    print(arr);

def check_prime(n) : 
    if n <= 1 : return False
    if n <= 3 : return True
    if n%2 == 0 or n%3 == 0 : return False
    for i in range(1, int(math.sqrt(n) +1), 6):
        if n%i == 0 or n%(i+2) == 0 : return False
    return True;

def gcd_hcf(a,b) : 
    if b == 0 : return a
    return gcd_hcf(b, a%b);

def lcm(a,b) : 
    return (a*b) // gcd_hcf(a,b);


print_all_divisors(n)

File: JS/BasicMaths/Maths.js
console.log("Running Maths in JavaScript");

function countDigits(n) {
  let count = 0;
  while (n > 0) {
    let lastDigit = n % 10;
    count++;
    n /= 10;
  }
  return count;
}

function reverseANumber(n) {
  let num = 0;
  while (n > 0) {
    let lastDigit = n % 10;
    num = num * 10 + lastDigit;
    n /= 10;
  }
  return num;
}

function checkPalindrome(n) {
  return n == reverseANumber(n);
}

function armstrongNumber(n) {
  let sum = 0,
    num = n;
  while (n > 0) {
    let lastDigit = n % 10;
    sum += Math.pow(lastDigit, 3);
    n /= 10;
  }
  return num == sum;
}

function printAllDivisors(n) {
  let arr = [];
  //for(let i = 1 ; i <= Math.sqrt(n) ; i++){
  for (let i = 1; i * i <= n; i++) {
    if (n % i == 0) {
      arr.push(i);
      if (i * i != n) arr.push(n / i);
    }
  }
  arr.sort();
  arr.forEach(function (num) {
    process.stdout.write(num + " ");
  });
}

function checkPrime(n) {
  if (n <= 1) return false;
  if (n <= 3) return true;
  if (n % 2 == 0 || n % 3 == 0) return false;
  for (let i = 5; i * i <= n; i += 6) {
    if (n % i == 0 || n % (i + 2) == 0) return false;
  }
  return true;
}

function gcdHcf(a,b) {
    if(b == 0) return a;
    return gcdHcf(b, a%b);
}

function lcm(a,b){ 
    return (a*b)/gcdHcf(a,b);
}
